### PR TITLE
Server spring

### PR DESCRIPTION
**Pull request by Zingerd**
Fix method LetterPositions

### CODE REVIEW

**Comment by MasterDjefer** [comment]
we should talk to determine what is wrong in your code

File: Project/src/main/java/app/hangman/Word.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    String word;
+
+    public String getWord() {
+        return word;
+    }
+
+    public String setWordByHash(String word) {
+        this.word = word;
+        return word;
+    }
```
Comment (MasterDjefer) [inline Project/src/main/java/app/hangman/Word.java]: why there is hash? why setter return smth?

File: Project/src/main/java/app/hangman/PositionWord.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package app.hangman;
+
+public class PositionWord {
+    int position;
+
+    public void setPosition(int position) {
+        this.position = position;
+    }
```
Comment (MasterDjefer) [inline Project/src/main/java/app/hangman/PositionWord.java]: method should insert new position into array
and change name for it

File: Project/src/main/java/app/hangman/WordController.java [hunk 3]
```diff
@@ -12,27 +13,60 @@
...
         word.add("LOVE");
+    }
+
+    static public String randomWord;
+
+    @RequestMapping(value = "/getRandomWord", method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public WordInfo controler() {
+
+        randomWord = word.stream().skip(random.nextInt(word.size())).findFirst().get();
```
Comment (MasterDjefer) [inline Project/src/main/java/app/hangman/WordController.java]: you dont need to save random word in local variable

File: Project/src/main/java/app/hangman/WordController.java [hunk 4]
```diff
@@ -12,27 +13,60 @@
...
 
         WordInfo exampleRestControler = new WordInfo();
         exampleRestControler.setWordHash(hashWord);
         exampleRestControler.setLetterCount(sizeWord);
 
         return exampleRestControler;
+    }
+
+    @RequestMapping(value = "/getWordByHash", method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (MasterDjefer) [inline Project/src/main/java/app/hangman/WordController.java]: this method should accept 1 param!!!  wordHash: int!!! (At least int, can be greater)

File: Project/src/main/java/app/hangman/WordController.java [hunk 5]
```diff
@@ -12,27 +13,60 @@
...
+    public Word WordByHash() {
+
+        Word word = new Word();
+        word.setWordByHash(randomWord);
+        return word;
+    }
```
Comment (MasterDjefer) [inline Project/src/main/java/app/hangman/WordController.java]: this method should access to hash table by input param wordHash and returns this word

File: Project/src/main/java/app/hangman/WordController.java [hunk 6]
```diff
@@ -12,27 +13,60 @@
...
+
+    @RequestMapping(value = "/getLetterPositions", method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public PositionWord LetterPositions(@RequestParam (value = "hashcode", required = true) Long hash,
+                                        @RequestParam (value = "letter", required = true) String letter)
```
Comment (MasterDjefer) [inline Project/src/main/java/app/hangman/WordController.java]: rename input params as I said

File: Project/src/main/java/app/hangman/PositionWord.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package app.hangman;
+
+import java.util.List;
+
+public class PositionWord {
+    List<Integer> position;
+
```
Comment (MasterDjefer) [inline Project/src/main/java/app/hangman/PositionWord.java]: add s

File: Project/src/main/java/app/hangman/WordController.java [hunk 8]
```diff
@@ -4,35 +4,93 @@
...
 public class WordController {
 
-    @RequestMapping(value = "/getRandomWord", method = RequestMethod.GET,
-            produces = MediaType.APPLICATION_JSON_VALUE)
-    public WordInfo controler() {
 
-        HashSet<String> word = new HashSet<String>();
-        Random random = new Random();
+    HashSet<String> word = new HashSet<String>();
+    Random random = new Random();
```
Comment (MasterDjefer) [inline Project/src/main/java/app/hangman/WordController.java]: add s

File: Project/src/main/java/app/hangman/PositionWord.java (hunk 2 above)
Comment (Zingerd) [reply Project/src/main/java/app/hangman/PositionWord.java]: I fixed it

File: Project/src/main/java/app/hangman/WordController.java (hunk 4 above)
Comment (Zingerd) [reply Project/src/main/java/app/hangman/WordController.java:44]: Okey

File: Project/src/main/java/app/hangman/WordController.java (hunk 6 above)
Comment (Zingerd) [reply Project/src/main/java/app/hangman/WordController.java]: Ok

File: Project/src/main/java/app/hangman/WordController.java (hunk 5 above)
Comment (Zingerd) [reply Project/src/main/java/app/hangman/WordController.java]: Ok, I fixed it

File: Project/src/main/java/app/hangman/WordController.java (hunk 4 above)
Comment (Zingerd) [reply Project/src/main/java/app/hangman/WordController.java]: Ok

File: Project/src/main/java/app/hangman/WordController.java (hunk 3 above)
Comment (Zingerd) [reply Project/src/main/java/app/hangman/WordController.java]: Ok

**Pull request merged**